src/fhirtypes/StructureDefinition.ts

```diff
@@ -225,6 +273,7 @@ export class StructureDefinition {
     let newMatchingElements: ElementDefinition[] = [];
     // Iterate over the path, filtering out elements that do not match
     for (const pathPart of parsedPath) {
+      // if we have all our elements can we do this with a tree instead
```

Comment:
Is this something for a future improvement? Or is this outdated now?

Reply:
This is something for a future improvement. I'll label it TODO to make that more apparent.